src/lsdb/core/crossmatch/kdtree_match.py

```diff
@@ -15,10 +16,42 @@ class KdTreeCrossmatch(AbstractCrossmatchAlgorithm):
 
     extra_columns = pd.DataFrame({"_DIST": pd.Series(dtype=np.dtype("float64"))})
 
+    def validate(self, n_neighbors: int = 1, radius_arcsec: float = 1, require_right_margin=True):
+        validate_radius(radius_arcsec)
+        if n_neighbors < 1:
+            raise ValueError("n_neighbors must be greater than 1")
+
+        # Check that we have the appropriate columns in our dataset.
+        if self.left.index.name != HIPSCAT_ID_COLUMN:
+            raise ValueError(f"index of left table must be {HIPSCAT_ID_COLUMN}")
+        if self.right.index.name != HIPSCAT_ID_COLUMN:
+            raise ValueError(f"index of right table must be {HIPSCAT_ID_COLUMN}")
+        column_names = self.left.columns
+        if self.left_metadata.catalog_info.ra_column not in column_names:
+            raise ValueError(f"left table must have column {self.left_metadata.catalog_info.ra_column}")
+        if self.left_metadata.catalog_info.dec_column not in column_names:
+            raise ValueError(f"left table must have column {self.left_metadata.catalog_info.dec_column}")
+
+        column_names = self.right.columns
+        if self.right_metadata.catalog_info.ra_column not in column_names:
+            raise ValueError(f"right table must have column {self.right_metadata.catalog_info.ra_column}")
+        if self.right_metadata.catalog_info.dec_column not in column_names:
+            raise ValueError(f"right table must have column {self.right_metadata.catalog_info.dec_column}")
```

Comment:
Could we make the method in the abstract class not abstract and use this as a default so a custom algorithm doesn't need to implement a validate?